custom_components/circadian_lighting/switch.py

```diff
@@ -331,22 +336,22 @@ def light_state_changed(self, entity_id, from_state, to_state):
         try:
             _LOGGER.debug(entity_id + " change from " + str(from_state) + " to " + str(to_state))
             if to_state.state == 'on' and from_state.state != 'on':
-                self.adjust_lights([entity_id], DEFAULT_INITIAL_TRANSITION)
+                self.adjust_lights([entity_id], CONF_INITIAL_TRANSITION)
         except:
             pass
 
     def sleep_state_changed(self, entity_id, from_state, to_state):
         try:
             _LOGGER.debug(entity_id + " change from " + str(from_state) + " to " + str(to_state))
             if to_state.state == self._sleep_state or from_state.state == self._sleep_state:
-                self.update_switch(DEFAULT_INITIAL_TRANSITION)
+                self.update_switch(CONF_INITIAL_TRANSITION)
         except:
             pass
     
     def disable_state_changed(self, entity_id, from_state, to_state):
         try:
             _LOGGER.debug(entity_id + " change from " + str(from_state) + " to " + str(to_state))
             if from_state.state == self._disable_state:
-                self.update_switch(DEFAULT_INITIAL_TRANSITION)
+                self.update_switch(CONF_INITIAL_TRANSITION)
```

Comment:
Can't do this. CONF_INITIAL_TRANSITION isn't an int, it's a configuration object. This _could_ be config.get(CONF_INITIAL_TRANSITION) but _should_ be:
```suggestion
        self.update_switch(self._initial_transition)
```